### PR TITLE
Add PR review checklist

### LINKED ISSUE #1169: Add PR review checklist
A PR review checklist should be added to assist when reviewing pull requests. This should take the form of 

- a concise checklist which is posted in the PR itself for the reviewer to tick off. This could be done e.g. by triggering a github action when a reviewer is assigned which comments the checklist in the PR
- an expanded form of the checklist in the TARDIS documentation which explains the points in more detail.

The checklist is not intended to be exhaustive (as different PRs will require different checks), but should give the reviewer a starting point for things to consider and provides a minimum standard which all PRs should meet before they are accepted.

### PR DESCRIPTION
**Pull request by arjunsavel**
**Per hackathon 7/15 discussion, GH actions currently don't pass secrets necessary for bot comments to PRs made from forks. Will convert to draft for now, but hopefully there'll be some progress on this front w.r.t. GH actions functionality at some point!**

## Description

I'm adding a GitHub workflow file that provides a comment below each PR after it's opened. 

## Motivation and Context

This fixes #1169 by providing a checklist for reviewers to reference for every opened PR.

## How Has This Been Tested?

I've run this successfully on [one of my own repos](https://github.com/arjunsavel/webb_scraping/pull/19).

## Screenshots (if appropriate):
comment_PR
**Per hackathon 7/15 discussion, GH actions currently don't pass secrets necessary for bot comments to PRs made from forks. Will convert to draft for now, but hopefully there'll be some progress on this front w.r.t. GH actions functionality at some point!

**Per hackathon 7/15 discussion, GH actions currently don't pass secrets necessary for bot comments to PRs made from forks. Will convert to draft for now, but hopefully there'll be some progress on this front w.r.t. GH actions functionality at some point!**

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have assigned/requested two reviewers for this pull request.

### CODE REVIEW

**Comment by arjunsavel** [comment]
@harpolea — what items do you think would be sufficient for the PR checklist? I think I remember you discussing this, but can't quite remember!

**Comment by harpolea** [comment]
@arjunsavel You can find the list that we came up with in the TARDIS google drive. I'll paste it here as well:

Before a PR is accepted, it must meet the following criteria:

- Is the necessary information provided?
    - Does the PR have a complete description? Does it explain what the PR is attempting to do/fix, does it explain how it does this?
    - Is there an explanation of why this PR is needed?
    - Please use the following PR template: https://github.com/tardis-sn/tardis/blob/master/.github/PULL_REQUEST_TEMPLATE.md
- Is this a duplicate PR?
    - If a new PR is clearly a duplicate, ask how this PR is different from the original PR? 
    - If this PR is about to be merged, close the original PR with a link to this new PR that solved the issue. 
- Does it pass existing tests and are new tests provided if required?
    - The test coverage should not decrease, and for new features should be at or very close to 100%.
- Is the code properly documented?
    - If this modifies existing code, then the docs should be updated. If this adds a new feature, additional documentation should be created.
    - Sphinx and docstrings in the code (in numpydoc format)
- Does this conform to PEP 8 and the TARDIS style guidelines?
    - TODO: write TARDIS style guidelines
- Does the PR fix the problem it describes?
    - Make sure it doesn’t e.g. just fix the problem for a specific case
    - Is this the best way of fixing the problem?
- Is the code tidy?
    - No unnecessary print lines or code comments

File: .github/workflows/PR_reviews.yml [hunk 1]
```diff
@@ -10,8 +10,23 @@ jobs:
...
+                  [ ] Please use the following PR template: https://github.com/tardis-sn/tardis/blob/master/.github/PULL_REQUEST_TEMPLATE.md \n
+                [ ] Is this a duplicate PR? \n
+                  [ ] If a new PR is clearly a duplicate, ask how this PR is different from the original PR? \n
+                  [ ] If this PR is about to be merged, close the original PR with a link to this new PR that solved the issue. \n
+                [ ] Does it pass existing tests and are new tests provided if required? \n
+                  [ ] The test coverage should not decrease, and for new features should be at or very close to 100%. \n
+                [ ] Is the code properly documented? \n
+                  [ ] If this modifies existing code, then the docs should be updated. If this adds a new feature, additional documentation should be created. \n
+                  [ ] Sphinx and docstrings in the code (in numpydoc format) \n
+                [ ] Does this conform to PEP 8 and the TARDIS style guidelines? \n
```
Comment (jaladh-singhal) [inline .github/workflows/PR_reviews.yml:26]: Where does "TARDIS style guidelines" live - can we add a link to them here?

Comment (arjunsavel) [reply .github/workflows/PR_reviews.yml:26]: I don't think that we have these yet — currently an issue (#1170 ) and a PR (#1183 ). Would you recommend removing this line for now, waiting until we have a style guide to push this, or something else?

**Comment by arjunsavel** [comment]
Also, the action is failing here, but it seems like this is [a problem](https://github.community/t/github-actions-are-severely-limited-on-prs/18179#M9249) with GitHub Actions permissions for PRs from forks. This should be fixed once it's merged with master. I've gotten something similar working on [one of my own repositories](https://github.com/arjunsavel/webb_scraping/pull/19).

File: .github/workflows/PR_reviews.yml (hunk 1 above)
Comment (jaladh-singhal) [reply .github/workflows/PR_reviews.yml]: Yeah I think we can remove TARDIS style guidelines and perhaps put `black` code style (with black hyperlinked).

File: .github/workflows/PR_reviews.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      - name: comment PR
+        uses: unsplash/comment-on-pr@master
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          msg: "Before a PR is accepted, it must meet the following criteria: \n
+                - [ ] Is the necessary information provided? \n
+                  \t - [ ] Does the PR have a complete description? Does it explain what the PR is attempting to do/fix, does it explain how it does this? \n
+                  \t - [ ] Is there an explanation of why this PR is needed? \n
+                  \t - [ ] Please use the following PR template: https://github.com/tardis-sn/tardis/blob/master/.github/PULL_REQUEST_TEMPLATE.md \n
```
Comment (jaladh-singhal) [inline .github/workflows/PR_reviews.yml:17]:
```suggestion
                  \t - [ ] Please use the [TARDIS PR template](https://github.com/tardis-sn/tardis/blob/master/.github/PULL_REQUEST_TEMPLATE.md) \n
```
I hope this doesn't break gh-action ?

**Comment by jaladh-singhal** [comment]
@epassaro @jordi5 I would encourage you to look at this - if the bot brings up the checklist, IMO we can remove the checklist altogether from PR template which will help to not mistaken them as to-dos by GH and will also make PR template simpler as other examples we saw today. Let me know what do you guys think?

File: .github/workflows/PR_reviews.yml (hunk 2 above)
Comment (arjunsavel) [reply .github/workflows/PR_reviews.yml:17]: Seems to work on my own repo!

**Comment by arjunsavel** [comment]
Per hackathon 7/15 discussion, GH actions currently don't pass secrets necessary for bot comments to PRs made from forks. Will convert to draft for now, but hopefully there'll be some progress on this front w.r.t. GH actions functionality at some point!

**Comment by epassaro** [comment]
@arjunsavel see https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_target

**Comment by marxwillia** [comment]
@aribalam This would be a good PR for you to work on during the bonding period.

**Comment by aribalam** [comment]
@marxwillia Right on it!

**Pull request closed without merging**